lib/lib/datamanagement/repository/grid_repository.dart

```diff
@@ -1,19 +1,23 @@
 import 'dart:convert';
 import 'package:f1_application/lib/model/driver.dart';
 import 'package:f1_application/lib/network/api/ergast_api_client.dart';
+import 'package:http/http.dart';
+import 'package:f1_application/lib/datamanagement/error_handler.dart';
 
 class GridRepository {
   GridRepository();
 
   final ErgastApiClient clientErgast = ErgastApiClient();
 
+  final ErrorHandler _errorHandler = ErrorHandler();

```

Comment:
Ez már így jó, viszont még egy olyan szépítést lehetne még mindig ezen csinálni, hogy ez az errorHandler mező és a catch ág ne a repositorykban legyen háromszor. 

Nem segítek ennél többet egyelőre, ha van ötleted beszéljünk róla. Az lenne a cél, hogy ez a függvény körülbelül így nézzen ki: 

```
Future<List<Driver>> fetchGrid(int seasonYear) async {
       performCall {
       Response response = await clientErgast.get(type: GetType.grid, year: seasonYear).timeout(const Duration(seconds: 10));
       Map<String, dynamic> jsonObject = jsonDecode(response.body);
       return Driver.listFromJson(jsonObject["MRData"]["DriverTable"]["Drivers"]);
       }
}
```

Reply:
Ezt majd még beszéljük át, mert a megoldásom, habár működik, és a tesztek is megfelelnek, de nem ez a legjobb módszer szerintem.